Cache element throws ClassCastException

### Current Behavior

Adapters with a `Cache` element that rely on `keyInputSessionKey` are broken.

Example of my Cache element that breaks.
`<Cache name="cache" keyInputSessionKey="cacheKey" timeToLiveSeconds="43200" timeToIdleSeconds="43200" />` 

<img width="1064" height="560" alt="Image" src="https://github.com/user-attachments/assets/18f48948-4cce-4c3c-ae33-1a4683906c6a" />

With the following exception:
```
org.frankframework.core.SenderException: exception calling JavaListener [Magdat_MetaDataSchemas_Cache]: (ClassCastException) class org.frankframework.stream.Message cannot be cast to class java.lang.String (org.frankframework.stream.Message is in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @7139bd31; java.lang.String is in module java.base of loader 'bootstrap')
	at org.frankframework.senders.IbisLocalSender.sendMessage(IbisLocalSender.java:286)
	at jdk.internal.reflect.GeneratedMethodAccessor674.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugSenderInputOutputAbort(IbisDebuggerAdvice.java:232)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugSenderSendMessage(IbisDebuggerAdvice.java:286)
	at jdk.internal.reflect.GeneratedMethodAccessor581.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.senders.IbisLocalSender$$SpringCGLIB$$0.sendMessage(<generated>)
	at org.frankframework.pipes.MessageSendingPipe.sendMessage(MessageSendingPipe.java:619)
	at org.frankframework.pipes.MessageSendingPipe.sendMessageWithRetries(MessageSendingPipe.java:413)
	at org.frankframework.pipes.MessageSendingPipe.doPipe(MessageSendingPipe.java:346)
	at org.frankframework.processors.CorePipeProcessor.processPipe(CorePipeProcessor.java:36)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.TrackPreviousPipeInMetadataProcessor.processPipe(TrackPreviousPipeInMetadataProcessor.java:42)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.CheckMessageSizePipeProcessor.processPipe(CheckMessageSizePipeProcessor.java:38)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LockerPipeProcessor.processPipe(LockerPipeProcessor.java:40)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.TransactionAttributePipeProcessor.executePipeProcess(TransactionAttributePipeProcessor.java:97)
	at org.frankframework.processors.TransactionAttributePipeProcessor.processPipe(TransactionAttributePipeProcessor.java:67)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LimitingParallelExecutionPipeProcessor.processPipe(LimitingParallelExecutionPipeProcessor.java:49)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LimitingParallelExecutionPipeProcessor.processPipe(LimitingParallelExecutionPipeProcessor.java:76)
	at jdk.internal.reflect.GeneratedMethodAccessor574.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeGetInputFrom(IbisDebuggerAdvice.java:188)
	at jdk.internal.reflect.GeneratedMethodAccessor573.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.processors.LimitingParallelExecutionPipeProcessor$$SpringCGLIB$$0.processPipe(<generated>)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.ExceptionHandlingPipeProcessor.processPipe(ExceptionHandlingPipeProcessor.java:40)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.InputOutputPipeProcessor.processPipe(InputOutputPipeProcessor.java:98)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.InputOutputPipeProcessor.processPipe(InputOutputPipeProcessor.java:236)
	at jdk.internal.reflect.GeneratedMethodAccessor572.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeInputOutputAbort(IbisDebuggerAdvice.java:160)
	at jdk.internal.reflect.GeneratedMethodAccessor571.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.processors.InputOutputPipeProcessor$$SpringCGLIB$$0.processPipe(<generated>)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.MonitoringPipeProcessor.processPipe(MonitoringPipeProcessor.java:40)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LogPipeProcessor.processPipe(LogPipeProcessor.java:57)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.CorePipeLineProcessor.processPipeLine(CorePipeLineProcessor.java:227)
	at jdk.internal.reflect.GeneratedMethodAccessor593.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeLineInputOutputAbort(IbisDebuggerAdvice.java:125)
	at jdk.internal.reflect.GeneratedMethodAccessor592.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.processors.CorePipeLineProcessor$$SpringCGLIB$$0.processPipeLine(<generated>)
	at org.frankframework.processors.LockerPipeLineProcessor.processPipeLine(LockerPipeLineProcessor.java:60)
	at org.frankframework.processors.TransactionAttributePipeLineProcessor.processPipeLine(TransactionAttributePipeLineProcessor.java:49)
	at org.frankframework.processors.LimitingParallelExecutionPipeLineProcessor.processPipeLine(LimitingParallelExecutionPipeLineProcessor.java:43)
	at org.frankframework.processors.CachePipeLineProcessor.processPipeLine(CachePipeLineProcessor.java:41)
	at org.frankframework.processors.InputOutputPipeLineProcessor.processPipeLine(InputOutputPipeLineProcessor.java:51)
	at org.frankframework.core.PipeLine.process(PipeLine.java:412)
	at org.frankframework.core.Adapter.processMessageWithExceptions(Adapter.java:833)
	at org.frankframework.receivers.Receiver.processMessageInAdapter(Receiver.java:1278)
	at org.frankframework.receivers.Receiver.processRequest(Receiver.java:1054)
	at org.frankframework.receivers.JavaListener.processRequest(JavaListener.java:185)
	at org.frankframework.receivers.JavaListener.processRequest(JavaListener.java:174)
	at org.frankframework.senders.IbisLocalSender.sendMessage(IbisLocalSender.java:279)
	at jdk.internal.reflect.GeneratedMethodAccessor674.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugSenderInputOutputAbort(IbisDebuggerAdvice.java:232)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugSenderSendMessage(IbisDebuggerAdvice.java:286)
	at jdk.internal.reflect.GeneratedMethodAccessor581.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.senders.IbisLocalSender$$SpringCGLIB$$0.sendMessage(<generated>)
	at org.frankframework.pipes.MessageSendingPipe.sendMessage(MessageSendingPipe.java:619)
	at org.frankframework.pipes.MessageSendingPipe.sendMessageWithRetries(MessageSendingPipe.java:413)
	at org.frankframework.pipes.MessageSendingPipe.doPipe(MessageSendingPipe.java:346)
	at org.frankframework.processors.CorePipeProcessor.processPipe(CorePipeProcessor.java:36)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.TrackPreviousPipeInMetadataProcessor.processPipe(TrackPreviousPipeInMetadataProcessor.java:42)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.CheckMessageSizePipeProcessor.processPipe(CheckMessageSizePipeProcessor.java:38)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LockerPipeProcessor.processPipe(LockerPipeProcessor.java:40)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.TransactionAttributePipeProcessor.executePipeProcess(TransactionAttributePipeProcessor.java:97)
	at org.frankframework.processors.TransactionAttributePipeProcessor.processPipe(TransactionAttributePipeProcessor.java:67)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LimitingParallelExecutionPipeProcessor.processPipe(LimitingParallelExecutionPipeProcessor.java:49)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LimitingParallelExecutionPipeProcessor.processPipe(LimitingParallelExecutionPipeProcessor.java:76)
	at jdk.internal.reflect.GeneratedMethodAccessor574.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeGetInputFrom(IbisDebuggerAdvice.java:188)
	at jdk.internal.reflect.GeneratedMethodAccessor573.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.processors.LimitingParallelExecutionPipeProcessor$$SpringCGLIB$$0.processPipe(<generated>)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.ExceptionHandlingPipeProcessor.processPipe(ExceptionHandlingPipeProcessor.java:40)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.InputOutputPipeProcessor.processPipe(InputOutputPipeProcessor.java:98)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.InputOutputPipeProcessor.processPipe(InputOutputPipeProcessor.java:236)
	at jdk.internal.reflect.GeneratedMethodAccessor572.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeInputOutputAbort(IbisDebuggerAdvice.java:160)
	at jdk.internal.reflect.GeneratedMethodAccessor571.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.processors.InputOutputPipeProcessor$$SpringCGLIB$$0.processPipe(<generated>)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.MonitoringPipeProcessor.processPipe(MonitoringPipeProcessor.java:40)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.AbstractPipeProcessor.lambda$processPipe$0(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.LogPipeProcessor.processPipe(LogPipeProcessor.java:57)
	at org.frankframework.processors.AbstractPipeProcessor.processPipe(AbstractPipeProcessor.java:46)
	at org.frankframework.processors.CorePipeLineProcessor.processPipeLine(CorePipeLineProcessor.java:227)
	at jdk.internal.reflect.GeneratedMethodAccessor593.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.support.AopUtils.invokeJoinpointUsingReflection(AopUtils.java:360)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.invokeJoinpoint(ReflectiveMethodInvocation.java:196)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.aspectj.MethodInvocationProceedingJoinPoint.proceed(MethodInvocationProceedingJoinPoint.java:102)
	at org.frankframework.ladybug.IbisDebuggerAdvice.debugPipeLineInputOutputAbort(IbisDebuggerAdvice.java:125)
	at jdk.internal.reflect.GeneratedMethodAccessor592.invoke(Unknown Source)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:569)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethodWithGivenArgs(AbstractAspectJAdvice.java:649)
	at org.springframework.aop.aspectj.AbstractAspectJAdvice.invokeAdviceMethod(AbstractAspectJAdvice.java:631)
	at org.springframework.aop.aspectj.AspectJAroundAdvice.invoke(AspectJAroundAdvice.java:71)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:173)
	at org.springframework.aop.interceptor.ExposeInvocationInterceptor.invoke(ExposeInvocationInterceptor.java:97)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:184)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:728)
	at org.frankframework.processors.CorePipeLineProcessor$$SpringCGLIB$$0.processPipeLine(<generated>)
	at org.frankframework.processors.LockerPipeLineProcessor.processPipeLine(LockerPipeLineProcessor.java:60)
	at org.frankframework.processors.TransactionAttributePipeLineProcessor.processPipeLine(TransactionAttributePipeLineProcessor.java:49)
	at org.frankframework.processors.LimitingParallelExecutionPipeLineProcessor.processPipeLine(LimitingParallelExecutionPipeLineProcessor.java:43)
	at org.frankframework.processors.CachePipeLineProcessor.processPipeLine(CachePipeLineProcessor.java:41)
	at org.frankframework.processors.InputOutputPipeLineProcessor.processPipeLine(InputOutputPipeLineProcessor.java:51)
	at org.frankframework.core.PipeLine.process(PipeLine.java:412)
	at org.frankframework.core.Adapter.processMessageWithExceptions(Adapter.java:833)
	at org.frankframework.receivers.Receiver.processMessageInAdapter(Receiver.java:1278)
	at org.frankframework.receivers.Receiver.processRequest(Receiver.java:1054)
	at org.frankframework.receivers.JavaListener.processRequest(JavaListener.java:185)
	at org.frankframework.receivers.JavaListener.processRequest(JavaListener.java:154)
	at nl.nn.adapterframework.dispatcher.DispatcherManagerImpl.processRequest(DispatcherManagerImpl.java:123)
	at org.frankframework.senders.IbisJavaSender.sendMessage(IbisJavaSender.java:133)
	at org.frankframework.larva.SenderThread.run(SenderThread.java:78)
Caused by: org.frankframework.core.ListenerException: (ClassCastException) class org.frankframework.stream.Message cannot be cast to class java.lang.String (org.frankframework.stream.Message is in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @7139bd31; java.lang.String is in module java.base of loader 'bootstrap')
	at org.frankframework.core.Adapter.processMessageWithExceptions(Adapter.java:836)
	at org.frankframework.receivers.Receiver.processMessageInAdapter(Receiver.java:1278)
	at org.frankframework.receivers.Receiver.processRequest(Receiver.java:1054)
	at org.frankframework.receivers.JavaListener.processRequest(JavaListener.java:185)
	at org.frankframework.receivers.JavaListener.processRequest(JavaListener.java:174)
	at org.frankframework.senders.IbisLocalSender.sendMessage(IbisLocalSender.java:279)
	... 250 more
Caused by: java.lang.ClassCastException: class org.frankframework.stream.Message cannot be cast to class java.lang.String (org.frankframework.stream.Message is in unnamed module of loader org.apache.catalina.loader.ParallelWebappClassLoader @7139bd31; java.lang.String is in module java.base of loader 'bootstrap')
	at org.frankframework.cache.AbstractCacheAdapter.transformKey(AbstractCacheAdapter.java:86)
	at org.frankframework.cache.AbstractCacheAdapter.transformKey(AbstractCacheAdapter.java:41)
	at org.frankframework.processors.CachePipeLineProcessor.processPipeLine(CachePipeLineProcessor.java:50)
	at org.frankframework.processors.InputOutputPipeLineProcessor.processPipeLine(InputOutputPipeLineProcessor.java:51)
	at org.frankframework.core.PipeLine.process(PipeLine.java:412)
	at org.frankframework.core.Adapter.processMessageWithExceptions(Adapter.java:833)
	... 255 more
```

### Expected Behavior

No exception

### Environment Information

9.4.2

### Steps To Reproduce

_No response_

### Configuration

```xml

```

### Input

_No response_

### What database are you using?

_No response_

### What browsers are you seeing the problem on?

_No response_

### Relevant Log Output

```console

```

### Anything else?

_No response_

--- a/core/src/main/java/org/frankframework/cache/AbstractCacheAdapter.java
+++ b/core/src/main/java/org/frankframework/cache/AbstractCacheAdapter.java
@@ -15,6 +15,8 @@
 */
 package org.frankframework.cache;
 
+import java.io.IOException;
+
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.Logger;
 import org.springframework.context.ApplicationContext;
@@ -82,8 +84,12 @@ public abstract class AbstractCacheAdapter<V> implements ICache<String,V>, Frank
 
 	@Override
 	public String transformKey(String input, PipeLineSession session) {
-		if (StringUtils.isNotEmpty(getKeyInputSessionKey()) && session!=null) {
-			input=(String)session.get(getKeyInputSessionKey());
+		if (StringUtils.isNotEmpty(getKeyInputSessionKey()) && session != null) {
+			try {
+				input = session.getMessage(getKeyInputSessionKey()).asString();
+			} catch (IOException e) {
+				log.error("{}cannot determine cache key", getLogPrefix(), e);
+			}
 		}
 		if (keyTp!=null) {
 			try {

--- a/core/src/main/java/org/frankframework/cache/AbstractCacheAdapter.java
+++ b/core/src/main/java/org/frankframework/cache/AbstractCacheAdapter.java
@@ -15,8 +15,6 @@
 */
 package org.frankframework.cache;
 
-import java.io.IOException;
-
 import org.apache.commons.lang3.StringUtils;
 import org.apache.logging.log4j.Logger;
 import org.springframework.context.ApplicationContext;
@@ -85,11 +83,7 @@ public abstract class AbstractCacheAdapter<V> implements ICache<String,V>, Frank
 	@Override
 	public String transformKey(String input, PipeLineSession session) {
 		if (StringUtils.isNotEmpty(getKeyInputSessionKey()) && session != null) {
-			try {
-				input = session.getMessage(getKeyInputSessionKey()).asString();
-			} catch (IOException e) {
-				log.error("{}cannot determine cache key", getLogPrefix(), e);
-			}
+			input = session.getString(getKeyInputSessionKey());
 		}
 		if (keyTp!=null) {
 			try {

